<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>rbims • rbims</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="rbims">
<meta property="og:description" content="rbims">
<meta name="twitter:card" content="summary_large_image">
<meta name="twitter:creator" content="@MirnaVRL">
<meta name="twitter:site" content="@MirnaVRL">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-000000-01"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-000000-01');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rbims</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/rRbiMs-vignette.html">rbims</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="rRbiMs-vignette_files/header-attrs-2.8/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>rbims</h1>
                        <h4 class="author">Mirna Vázquez Rosas Landa, Valerie de Anda Torres, Sahil Shan, and Brett Baker</h4>
            
            <h4 class="date">2021-05-18</h4>
      
      
      <div class="hidden name"><code>rRbiMs-vignette.Rmd</code></div>

    </div>

    
    
<p>rRbiMs is a user-friendly R workflow to infer metabolic potentials from unannotated genomes in an accurate and reproducible manner.</p>
<p>Most annotation pipelines use predefined databases to add the functional annotation. However, they do not allow the user to add metadata information, making it challenging to explore the metabolism based on a specific scientific question. Our motivation to build the package rRbiMs is to create a workflow that helps researchers to explore metabolism data in a reproducible manner. rRbiMs reads different database outputs and includes a new custom-curated database that the user can easily access. Its module-based design allows the user to choose between running the whole pipeline or just part of it. A key feature is that it facilitates the incorporation of metadata such as taxonomy and sampling data, allowing the user to answer specific questions of biological relevance.</p>
<p>The standard pipeline consists of four steps: read, mapping, sub-setting, and graphic plotting. In a typical workflow, the read and mapping steps utilize the output from KEGG to create a profile-abundance matrix organized by pathways. After mapping, the metadata can be added and used within the sub-setting step, which also allows multivariate analysis. Finally, the plot steps allow the generation of publication-quality figures and the exploration of data through custom visualization.</p>
<div id="kegg-based-profile" class="section level2">
<h2 class="hasAnchor">
<a href="#kegg-based-profile" class="anchor"></a>KEGG-based profile</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">rRbiMs</span><span class="op">)</span></code></pre></div>
<p>Read the KofamKOALA/KofamScan output.</p>
<p>The function that rRbiMs use to read the input table is <code>read_ko</code>. And it reads the raw output from the KofamKOALA/KofamScan programs.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">?</span><span class="va">read_ko</span></code></pre></div>
<p><code>rRbiMs</code> contains external data that allow us to test this function. You can load the file from here (<a href="https://github.com/mirnavazquez/RbiMs/blob/main/inst/extdata/KEGG_bins.txt" class="uri">https://github.com/mirnavazquez/RbiMs/blob/main/inst/extdata/KEGG_bins.txt</a>) and try.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ko_bin_table</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/read_ko.html">read_ko</a></span><span class="op">(</span><span class="st">"KEGG_bins.txt"</span><span class="op">)</span></code></pre></div>
<p>The read_ko function will create a table that contains the abundance of each KO within each bin.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">ko_bin_table</span><span class="op">)</span></code></pre></div>
<p>Now we will use the function mapping_ko to map the KO ids to the main KEGG-based database. For now it just includes the KEGG database and DiTing database (<a href="https://github.com/xuechunxu/DiTing" class="uri">https://github.com/xuechunxu/DiTing</a>).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">?</span><span class="va">mapping_ko</span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ko_bin_mapp</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/mapping_ko.html">mapping_ko</a></span><span class="op">(</span><span class="va">ko_bin_table</span><span class="op">)</span></code></pre></div>
<p>This will create a profile-abundance matrix organized by pathways.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">ko_bin_mapp</span><span class="op">)</span></code></pre></div>
</div>
<div id="profile-subsetting" class="section level2">
<h2 class="hasAnchor">
<a href="#profile-subsetting" class="anchor"></a>Profile subsetting</h2>
<p>Currently there are two functions to help you understand the profile, <code>get_subset_pca</code> and <code>get_subset_unique</code>. The first function makes a principal component analysis to identify the most important KO in the data set, while the second function use the metadata information to identify KO that are only present in certain group of bins.</p>
<p>First, lets load the metadata.</p>
<p>To read the metadata we will use the function <code>read_excel</code> from the package <code>readxl</code>, if you dont have it install it, please do. Then you can download the file from here: <a href="https://github.com/mirnavazquez/RbiMs/blob/main/inst/extdata/metadata.xlsx" class="uri">https://github.com/mirnavazquez/RbiMs/blob/main/inst/extdata/metadata.xlsx</a>, and try.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">metadata</span><span class="op">&lt;-</span><span class="fu">read_excel</span><span class="op">(</span><span class="st">"metadata.xlsx"</span><span class="op">)</span></code></pre></div>
<p>Now lets consider a case where we have bins from two sampling sites, water column and sediments and we want to know which KO are only present in the bins found in the water column?</p>
<p>To do so we can use the function <code>get_subset_unique</code> and the metadata information like this:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">?</span><span class="va">get_subset_unique</span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ko_subsett</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/get_subset_unique.html">get_subset_unique</a></span><span class="op">(</span><span class="va">ko_bin_mapp</span>, <span class="va">metadata</span>, <span class="va">Sample_site</span>, <span class="st">"Water_column"</span><span class="op">)</span></code></pre></div>
<p>If you check the dimention of this new object and compare it with the original mapping object, you may be able to see the difference.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">ko_subsett</span><span class="op">)</span>     
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">ko_bin_mapp</span><span class="op">)</span>  </code></pre></div>
</div>
<div id="profile-plots" class="section level2">
<h2 class="hasAnchor">
<a href="#profile-plots" class="anchor"></a>Profile plots</h2>
<p><code>rRbiMs</code> for now support two types of analysis for visualization, <code>Raw abundance</code> and <code>Relative abundance</code>, this two analysis can be plot in <code>bubble plots</code> or <code>heatmaps</code>.</p>
<p>You can use the funtion colnames to know what type of arguments you can use within the functios.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ko_bin_mapp</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span></code></pre></div>
<p>The <code>Raw abundance</code> analysis show the abundance of each metabolism in an specific genome.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/rRbiMs/man/plot_bubble_abundance.html">plot_bubble_abundance</a></span><span class="op">(</span><span class="va">ko_bin_mapp</span>, <span class="va">metadata</span>, <span class="va">Bin_name</span>, <span class="va">Cycle</span>, <span class="va">Genus</span><span class="op">)</span></code></pre></div>
<p>While <code>Relative abundance</code> analysis normalize each metabolism by the number of KO componing an specific metabolism.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/rRbiMs/man/plot_bubble_percentage.html">plot_bubble_percentage</a></span><span class="op">(</span><span class="va">ko_bin_mapp</span>, <span class="va">metadata</span>, <span class="va">Bin_name</span>, <span class="va">Cycle</span>, <span class="va">Genus</span><span class="op">)</span></code></pre></div>
<p>The heatmap functions work similar, but they allow to show better the metadata.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/rRbiMs/man/plot_heatmap_percentage.html">plot_heatmap_percentage</a></span><span class="op">(</span><span class="va">ko_bin_mapp</span>, <span class="va">metadata</span>, <span class="va">KO</span>, <span class="va">Cycle</span>, <span class="va">Species</span><span class="op">)</span> </code></pre></div>
<p>In the case of the abundance function, you will be always looking at the abundance of KO and the with the annotation you will be able to see how is this KO related to other categories of metabolism.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/rRbiMs/man/plot_heatmap_abundance.html">plot_heatmap_abundance</a></span><span class="op">(</span><span class="va">ko_bin_mapp</span>, <span class="va">metadata</span>, <span class="va">Cycle</span>, <span class="va">Species</span><span class="op">)</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Mirna Vázquez-Rosas-Landa.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
